python/cudf/cudf/utils/cudautils.py

```diff
@@ -15,223 +16,29 @@
-            gpu_copy.forall(arr.size)(arr, out)
-    return out
-
-
 def as_contiguous(arr):
     assert arr.ndim == 1
-    out = rmm.device_array(shape=arr.shape, dtype=arr.dtype)
-    return copy_array(arr, out=out)
-
-
-# Copy column into a matrix
-
-
-@cuda.jit
-def gpu_copy_column(matrix, colidx, colvals):
-    tid = cuda.grid(1)
-    if tid < colvals.size:
-        matrix[colidx, tid] = colvals[tid]
-
-
-def copy_column(matrix, colidx, colvals):
-    assert matrix.shape[1] == colvals.size
-    if colvals.size > 0:
-        configured = gpu_copy_column.forall(colvals.size)
-    configured(matrix, colidx, colvals)
+    cupy_dtype = arr.dtype
+    if np.issubdtype(cupy_dtype, np.datetime64):
+        cupy_dtype = np.dtype("int64")
+        arr = arr.view("int64")
+    out = cupy.ascontiguousarray(cupy.asarray(arr))
+    return cuda.as_cuda_array(out).view(arr.dtype)
 
 
 # Mask utils
 
 
-@cuda.jit
-def gpu_set_mask_from_stride(mask, stride):
-    bitsize = mask_bitsize
-    tid = cuda.grid(1)
-    if tid < mask.size:
-        base = tid * bitsize
-        val = 0
-        for shift in range(bitsize):
-            idx = base + shift
-            bit = (idx % stride) == 0
-            val |= bit << shift
-        mask[tid] = val
-
-
-def set_mask_from_stride(mask, stride):
-    taskct = mask.size
-    if taskct > 0:
-        configured = gpu_set_mask_from_stride.forall(taskct)
-        configured(mask, stride)
-
-
-@cuda.jit
-def gpu_copy_to_dense(data, mask, slots, out):
-    tid = cuda.grid(1)
-    if tid < data.size and mask_get(mask, tid):
-        idx = slots[tid]
-        out[idx] = data[tid]
-
-
-@cuda.jit
-def gpu_invert_value(data, out):
-    tid = cuda.grid(1)
-    if tid < data.size:
-        out[tid] = not data[tid]
-
-
-def invert_mask(arr, out):
-    if arr.size > 0:
-        gpu_invert_value.forall(arr.size)(arr, out)
-
-
-def fill_mask(data, mask, value):
-    """fill a column with the same value using a custom mask
-
-    Parameters
-    ----------
-    data : device array
-        data
-    mask : device array
-        validity mask
-    value : scale
-        fill value
-
-    Returns
-    -------
-    device array
-        mask filled column with scalar value
-    """
-
-    out = rmm.device_array_like(data)
-    out.copy_to_device(data)
-    if data.size > 0:
-        configured = gpu_fill_masked.forall(data.size)
-        configured(value, mask, out)
-    return out
-
-
-@cuda.jit
-def gpu_fill_value(data, value):
-    tid = cuda.grid(1)
-    if tid < data.size:
-        data[tid] = value
-
-
-def fill_value(arr, value):
-    """Fill *arr* with value
-    """
-    if arr.size > 0:
-        gpu_fill_value.forall(arr.size)(arr, value)
-
-
 def full(size, value, dtype):
-    out = rmm.device_array(size, dtype=dtype)
-    fill_value(out, value)
-    return out
+    cupy_dtype = dtype
+    if np.issubdtype(cupy_dtype, np.datetime64):
+        cupy_dtype = np.int64
+
+    out = cupy.full(size, value, cupy_dtype)
```

Comment:
Need to handle the situation here where `value` is a numpy scalar or some other `datetime64` scalar that cupy can't handle.

Reply:
Note this will also be a moot point with bindings to #4352